profiles/api/oauth2.py

```diff
@@ -122,9 +125,16 @@ def _token() -> Response:
             raise ValueError('Got token_type {}, expected Bearer'.format(
                 json_data.get('token_type')))
 
-        filtered_json_data = {your_key: json_data[your_key] for your_key in
-                              ['access_token', 'expires_in', 'name', 'orcid', 'token_type']}
+        json_data = {key: json_data[key] for key in
+                     ['access_token', 'expires_in', 'name', 'orcid', 'token_type']}
 
-        return make_response(jsonify(filtered_json_data), response.status_code)
+        try:
+            profile = profiles.get_by_orcid(json_data['orcid'])
+            profile.name = json_data['name']
+        except ProfileNotFound:
+            profile = Profile(profiles.next_id(), json_data['name'], json_data['orcid'])
```

Comment:
is the session automatically committed (assuming there is an open transaction) at the end of the request? when in particular?

Reply:
End of the request (`SQLALCHEMY_COMMIT_ON_TEARDOWN`). It's a bit of a controversial technique, but transactions are hard with repositories. (Never found a good solution.)

It's probably acceptable here, since it does actually do that much.

Reply:
We'll go for an explicit Unit Of Work if needed